src/client/serialize/daos_serialize.c

```diff
@@ -8,13 +8,89 @@
 
 /* for user attr dataset */
 struct usr_attr {
-	char	*attr_name;
-	hvl_t	attr_val;
+    char    *attr_name;
+    hvl_t   attr_val;
 };
 
+/* struct obj_id type defined for hdf5 attribute. The DAOS_PROP_CO_ROOTS
+ * points to an array of daos_obj_id_t, which is a struct that contains
+ * hi, lo uint64_t's. In order to write these to an hdf5 file a similar
+ * type has to be defined to describe/store it to hdf5
+ */
+struct obj_id {
+    uint64_t hi;
+    uint64_t lo;
+};
+
+int
+serialize_roots(hid_t file_id, struct daos_prop_entry *entry, const char *prop_str)
+{
+	int				rc = 0;
+	hid_t				status = 0;
+	struct daos_prop_co_roots	*roots;
+	struct obj_id			root_oids[4];
+	hsize_t				attr_dims[1];
+	/* HDF5 returns non-negative identifiers if successfully opened/created */
+	hid_t				attr_dtype = -1;
+	hid_t				attr_dspace = -1;
+	hid_t				usr_attr = -1;
+	int				i = 0;
+
+	if (entry == NULL || entry->dpe_val_ptr == NULL) {
+        	D_GOTO(out, rc = -DER_INVAL);
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs